components/commander/common/commander_url.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/commander/common/commander_url.h"
+
+#include <string>
+#include <vector>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/unguessable_token.h"
+
+namespace commander {
+namespace {
+// Gets the scheme for brave-command urls. This contains an UnguessableToken
+// which is regenerated per-launch as part of a defence-in-depth strategy.
+std::string GetScheme() {
+  static std::string scheme = base::ToLowerASCII(
+      "brave-commands-" + base::UnguessableToken::Create().ToString());
+  return scheme;
+}
+}  // namespace
+
+GURL GetCommandURL(uint32_t command_index, uint32_t result_set_id) {
+  return GURL(GetScheme() + std::string("://") +
+              base::NumberToString(command_index) + "/" +
+              base::NumberToString(result_set_id));
+}
+
+bool TryParseCommandURL(const GURL& url,
+                        uint32_t* command_index,
+                        uint32_t* result_set_id) {
+  DCHECK(command_index && result_set_id);
+  if (!url.SchemeIs(GetScheme())) {
```

Comment:
I'm not sure but GetScheme() may not work as UnguessableToken() will create new token on this call. If so, I think we shouls use RE2(regex lib used in C++ layer), or extract scheme part and check it.

Reply:
Shouldn't it be okay, as the scheme in `GetScheme` is in a static variable (or is their some C++ madness that I'm going to shoot myself in the foot with :laughing: )

Reply:
Ah, yeah, my bad. You're absolutely right!